### PR TITLE
Simplifying how build works

### PR DESCRIPTION
**Pull request by billonahill**
Getting into the code to work on a possible patch I made some changes to simplify how the build works. You can now build all targets with just two params:

ant -Dhadoop.dir=... -Dpig.dir=... jar

Or similarly from make. The other params now have sensible defaults.

I changed the sub-project to create the jars under their build dir by default, instead of the root directory. I also added error messaging if required params are missing or invalid.

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
     <property environment="env"/>
 	<condition property="ispig07">
-		<contains string="${pig.version}" substring="0.7"/>
+		<contains string="${pig.jar}" substring="pig-0.7"/>
```
Comment (dvryaboy) [inline build.xml:13]: Not related to this pull request, but wow that's an old version. Let's make a note to bump this to 0.11 once it's released. Probably doesn't make sense to bump to 0.9 right now (the existing code does work with 0.9 and 0.11 in practice).

Comment (billonahill) [inline build.xml:13]: This is just for the ispig07 check, which is used to decide if squeal should be built or not. AFAICT, squeal has not been upgraded to be compatible with Pig 0.9 and beyond. Yes, very old indeed.

**Comment by vertica** [comment]
Changes look good. I pulled it and it built after I included pig.jar=... 
I'll include logic to infer the jar from pig.dir.

On 01/22/2013 03:27 AM, Bill Graham wrote:

> Getting into the code to work on a possible patch I made some changes 
> to simplify how the build works. You can now build all targets with 
> just two params:
> 
> ant -Dhadoop.dir=... -Dpig.dir=... jar
> 
> Or similarly from make. The other params now have sensible defaults.
> 
> I changed the sub-project to create the jars under their build dir by 
> default, instead of the root directory. I also added error messaging 
> if required params are missing or invalid.
> 
> ---
> 
> ```
>     You can merge this Pull Request by running
> ```
> 
>    git pull https://github.com/billonahill/Vertica-Hadoop-Connector improve_build
> 
> Or view, comment on, or merge it at:
> 
> https://github.com/vertica/Vertica-Hadoop-Connector/pull/2
> 
> ```
>     Commit Summary
> 
> * Cleaning up how build works so you only need to specify
>   hadoop.dir and pig.jar when running ant targets. Changing
>   default build location to not be the root dir.
> 
> 
>     File Changes
> 
> * *M* build.xml (2)
> * *M* hadoop-connector/build.xml (30)
> * *M* hadoop-example/build.xml (27)
> * *M* makefile (2)
> * *M* pig-connector/build.xml (44)
> * *M* squeal/build.xml (38)
> 
> 
>     Patch Links:
> 
> * https://github.com/vertica/Vertica-Hadoop-Connector/pull/2.patch
> * https://github.com/vertica/Vertica-Hadoop-Connector/pull/2.diff
> ```

## 

Stephen Walkauskas      | Software Developer | Vertica, an HP company
swalkauskas@vertica.com | +1.617.596.9170    | www.vertica.com

The Vertica Analytic Database.  Simply Fast…

**Comment by billonahill** [comment]
Thanks Stephen. The pig.dir was a typo in my original comment. I meant to say pig.jar is all that is needed, so I think we're good to go here. hadoop.dir and pig.jar are required.

**Comment by swalkaus** [comment]
Yes, I realized that when I went to make the change:)

On 01/27/2013 12:47 PM, Bill Graham wrote:

> Thanks Stephen. The pig.dir was a typo in my original comment. I meant 
> to say pig.jar is all that is needed, so I think we're good to go 
> here. hadoop.dir and pig.jar are required.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/vertica/Vertica-Hadoop-Connector/pull/2#issuecomment-12757989. 

## 

Stephen Walkauskas      | Software Developer | Vertica, an HP company
swalkauskas@vertica.com | +1.617.596.9170    | www.vertica.com

The Vertica Analytic Database.  Simply Fast…

**Pull request merged**